packages/layout/src/text/processSmallCaps.js

```diff
@@ -27,56 +27,38 @@ const processSmallCaps = fragments => {
         processedFragments.push({ string, attributes });
         return;
       }
-      const words = string.split(' ');
+
+      const words = string.split(/(?=[A-Z][a-z])|(?=[A-Z])/) || [];
+
       const processedWordsFragments = words
         .map(word => {
-          const hasCaps = hasCapitals(word);
-          if (hasCaps) {
-            const chars = word.split('');
-
-            let tempStr = '';
-            const substrings = [];
-            chars.map(char => {
-              if (hasCapitals(char)) {
-                substrings.push(tempStr);
-                substrings.push(char);
-                tempStr = '';
-                return char;
-              }
-              tempStr += char;
-              return char;
+          const chars = word.split('') || [];
+          let uprecaseString = '';
+          let lowercaseString = '';
+          const stringFragments = [];
+          chars.map(char => {
```

Comment:
shall we use forEach instead of map here?

Reply:
done